src/Printers/Communication/UsbPrinterDeviceChannel.ts

```diff
@@ -0,0 +1,203 @@
+import { WebZlpError } from '../../WebZlpError';
+import { LineBreakTransformer } from './LineBreakTransformer';
+import {
+    PrinterChannelType,
+    IPrinterDeviceChannel,
+    PrinterCommMode,
+    PrinterCommunicationError
+} from './PrinterCommunication';
+
+/** Class for managing the WebUSB communication with a printer. */
+export class UsbPrinterDeviceChannel extends EventTarget implements IPrinterDeviceChannel {
+    private device: USBDevice;
+    private deviceIn: USBEndpoint;
+    private deviceOut: USBEndpoint;
+
+    public enableConsoleDebug = false;
+
+    private _commMode: PrinterCommMode;
+    public get commMode() {
+        return this._commMode;
+    }
+
+    get channelMode(): PrinterChannelType {
+        return PrinterChannelType.usb;
+    }
+
+    private _ready: Promise<boolean>;
+    public get ready(): Promise<boolean> {
```

Comment:
This is a little worrysome, returning a promise (that isn't new) means it can be manipulated/resolved by a consumer and then when you call for it again its in a bad state. I would either make sure this is always a new promise (or given that its meant to signal ready-ness, would just return a simple boolean.)